app/views/users/new.html.erb

```diff
@@ -1,3 +1,23 @@
-<% provide :title, t(".sign_up") %>
-<h1><%= t ".sign_up" %></h1>
-<p><%= t ".this_will_be_a_signup_page_for_new_users" %></p>
+<% provide :title, t("sign_up") %>
+<h1><%= t "sign_up" %></h1>
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for @user, url: signup_path do |f| %>
```

Comment:
tìm hiểu cho a sự khác nhau giữa form_for và form_tag nhé, chỗ này dùng form_tag có sinh ra được form giống form_for sinh ra không?

Reply:
form for thì tạo form khi cần có một đối tượng ạ
form tag thì tạo form khi không cần đối tượng 
chỗ này dùng form tag thì không giống form for vì ở đây chúng ta sử dụng biến @user để tuơng tác với database

Reply:
tìm hiểu lại xem nhé.

Reply:
form_for thường xuyên sử dụng nó để tương tác trực tiếp với models  mà chúng ta đã tạo trong database.  Tuy nhiên trong khá nhiều trường hợp chúng ta cần tạo một form mà không cần tương tác với database, nó không nhất thiết phải kết nối trực tiếp với bất kỳ model nào, form_tag giúp chúng ta làm được điều này. Và cả 2 đều có thể thay thế cho nhau ạ. Trong form_for, tham số đầu tiên chính là một biến instance sinh ra từ trong controller hoặc có thể là một record. Còn trong form_tag, tham số đầu tiên chính là một path, nếu không set gía trị, method sẽ được chọn mặc định là 'post'.
và sau đó tạo các trường với text_field_tag, select_tag, submit_tag... thay vì f.text_field, f.select, f.submit... như trong form_for

Reply:
dùng form_tag có sinh ra được form giống form_for sinh ra không?

Reply:
Có sinh ra dc ạ

Reply:
làm cách nào hả e?

Reply:
dùng <%= form_tag "/user" do %> ạ
giả sử như trong code của e 
<%= form_tag "/user" do %>
   <%= label_tag :name %>
  <%= text_field_tag :name %>
   <%= submit_tag %>
<% end %>

Reply:
e inspect xem đoạn html form bằng 2 cách sinh ra có giống nhau k nhé

Reply:
sẽ sinh ra nhưng phải thủ công ạ
ví dụ như <%= form_tag signup_path do %>
   <%= label_tag :name %>
   <%= text_field(:user, :name) %>
   <%= submit_tag %>
<% end %>

Reply:
giờ e thử inspect element cái form do form_for sinh ra rồi e dùng form_tag sinh ra form làm sao cho inspect element thấy form sinh ra nó giống form do form_for sinh ra nhé

Reply:
![screenshot from 2018-02-02 17-09-08](https://user-images.githubusercontent.com/22794404/35727777-d6f5adda-083b-11e8-9f63-63dcd6dd8d84.png)
form tag 
![screenshot from 2018-02-02 17-09-20](https://user-images.githubusercontent.com/22794404/35727779-d729716a-083b-11e8-994f-99e07f25af2e.png)
form for
cả 2 cách đều có thể đăng kí như bình thường ạ

Reply:
e dùng form_tag như thế nào để sinh ra được form như thế?

Reply:
![screenshot from 2018-02-02 17-10-48](https://user-images.githubusercontent.com/22794404/35728988-e649f77e-083f-11e8-8046-530e19322a8c.png)
code của e giữa form tag và form for đây ạ
@